mlink/mlink-functional-tests.groovy

```diff
@@ -0,0 +1,107 @@
+library changelog: false, identifier: "lib@master", retriever: modernSCM([
+        $class: 'GitSCMSource',
+        remote: 'https://github.com/Percona-Lab/jenkins-pipelines.git'
+])
+
+pipeline {
+    agent {
+        label 'master'
+    }
+    environment {
+        PATH = '/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/ec2-user/.local/bin'
+    }
+    parameters {
+        string(name: 'PSMDB', defaultValue: 'percona/percona-server-mongodb', description: 'PSMDB docker image')
+        string(name: 'MLINK_BRANCH', defaultValue: 'main', description: 'Mongo Link Branch')
+        string(name: 'GO_VER', defaultValue: 'latest', description: 'GOLANG docker image for building PBM from sources')
+        choice(name: 'instance', choices: ['docker-64gb','docker-64gb-aarch64'], description: 'Ec2 instance type for running tests')
+        string(name: 'PSMDB_TESTING_BRANCH', defaultValue: 'main', description: 'psmdb-testing repo branch')
+    }
+    stages {
+        stage('Set build name'){
+            steps {
+                script {
+                    currentBuild.displayName = "${params.MLINK_BRANCH}-${params.PSMDB}"
+                }
+            }
+        }
+        stage ('Run tests') {
+            matrix {
+                agent {
+                    label "${params.instance}"
+                }
+                axes {
+                    axis {
+                        name 'TEST'
+                        values 'replicaset', 'shard'
```

Comment:
For the nearest future mlink would support RS setup only, so there is no need for now to divide runs based on setup type. I think it's more practical to run the same set of tests for different PSMDB versions, like here https://github.com/Percona-Lab/jenkins-pipelines/blob/master/pbm/pbm-functional-tests-full.groovy#L37-L40